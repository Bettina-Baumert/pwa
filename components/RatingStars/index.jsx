/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import times from 'lodash/times';
import StarIcon from 'Components/icons/StarIcon';
import StarHalfIcon from 'Components/icons/StarHalfIcon';
import styles from './style';
import { RATING_SCALE_DIVISOR } from './constants';

/**
 * The available style keys for the rating stars.
 */
const availableStyles = styles.iconStyles;

/**
 * The rating stars component.
 * @param {Object} props The component props.
 * @returns {JSX}
 */
class RatingStars extends React.Component {
  static propTypes = {
    value: PropTypes.number.isRequired,
    className: PropTypes.string,
    display: PropTypes.oneOf(Object.keys(availableStyles)),
  };

  static defaultProps = {
    className: '',
    display: 'small',
  };

  /**
   * Only update the component if the star rating changed.
   * @param {Object} nextProps The next component props.
   * @returns {boolean}
   */
  shouldComponentUpdate(nextProps) {
    return nextProps.value !== this.props.value;
  }

  /**
   * Renders the component.
   * @returns {JSX}
   */
  render() {
    const { value } = this.props;
    const numStars = 5;
    const ratedStars = value / RATING_SCALE_DIVISOR;
    const numFullStars = Math.floor(ratedStars);
    const numHalfStars = Math.ceil(ratedStars - numFullStars);

    const size = styles.iconStyles[this.props.display].iconSize;

    const className = [styles.container, this.props.className].join(' ');
    const iconClassName = [styles.iconStyles[this.props.display].iconStyle, styles.icon].join(' ');

    const emptyStars = [
      ...times(numStars, i =>
        <div className={iconClassName} key={i}>
          <StarIcon size={size} />
        </div>
      ),
    ];
    const filledStars = [
      ...times(numFullStars, i =>
        <div className={iconClassName} key={i}>
          <StarIcon size={size} />
        </div>
      ),
      ...times(numHalfStars, i =>
        <div className={iconClassName} key={i + numFullStars}>
          <StarHalfIcon size={size} />
        </div>
      ),
    ];

    return (
      <div className={className}>
        <div className={styles.emptyStars}>
          {emptyStars}
        </div>
        <div className={styles.filledStars}>
          {filledStars}
        </div>
      </div>
    );
  }
}

export default RatingStars;
