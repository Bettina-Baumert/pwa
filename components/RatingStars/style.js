/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import css from 'glamor';
import colors from 'Styles/colors';

const container = css({
  position: 'relative',
}).toString();

const icon = css({
  display: 'inline-block',
  verticalAlign: 'top',
}).toString();

const emptyStars = css({
  color: colors.shade7,
}).toString();

const filledStars = css({
  position: 'absolute',
  color: colors.primary,
  top: 0,
}).toString();

/**
 * The rating stars styles that can be selected by passing the style key to the
 * @type {Object}
 */
const iconStyles = {
  small: {
    iconSize: '1em',
    iconStyle: css({
      marginRight: 0,
    }).toString(),
  },
  big: {
    iconSize: '1.24em',
    iconStyle: css({
      marginRight: '0.25em',
    }).toString(),
  },
};

export default {
  container,
  icon,
  iconStyles,
  emptyStars,
  filledStars,
};
