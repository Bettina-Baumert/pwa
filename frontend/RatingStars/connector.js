/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { connect } from 'react-redux';
import { getProductById } from '@shopgate/pwa-common-commerce/product/selectors/product';

/**
 * Maps the current application state to the component props.
 * @param {Object} state The current application state.
 * @param {Object} props The current component props.
 * @return {Object} The populated component props.
 */
const mapStateToProps = (state, props) => ({
  product: getProductById(state, props.id).productData || null,
});

export default connect(mapStateToProps);
