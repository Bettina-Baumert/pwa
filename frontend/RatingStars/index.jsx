/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import appConfig from '@shopgate/pwa-common/helpers/config';
import Stars from './components/Stars';
import connect from './connector';

const { hasReviews } = appConfig;

/**
 * The RatingStars component.
 * @return {JSX}
 */
const RatingStars = ({ product }) => {
  if (!hasReviews || !product || !product.rating || product.rating.count === 0) {
    return null;
  }

  return <Stars value={product.rating.average} />;
};

RatingStars.propTypes = {
  product: PropTypes.shape().isRequired,
};

export default connect(RatingStars);
