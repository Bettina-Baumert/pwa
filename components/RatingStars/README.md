# RatingStars
---
The RatingStars component is a simple component that displays five stars based on a given rating value.

## Getting Started

```markup
import { RatingStars } from 'Templates/components';

<RatingStars value={0.5} />
```

## Props

### value (required)

_Type_: `number  

The rating value that displays stars accordingly. It needs to be a floating number [0...1].

```
// Displays 4 filled stars and 1 empty star.
<RatingStars value={0.8} />
```

### display

_Type_: `'small'`, `'big'`, `'large'`, any custom defined style key.<br>
_Defaults_: `'small'` or first style key defined.<br>

The display style of the stars. 

```
// Display a bigger version of the rating stars.
<RatingStars value={0.8} display="big" />
```

