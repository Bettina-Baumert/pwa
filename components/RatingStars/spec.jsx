/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { shallow, mount } from 'enzyme';
import StarIcon from 'Components/icons/StarIcon';
import StarHalfIcon from 'Components/icons/StarHalfIcon';
import RatingStars from './index';

const numEmptyStars = 5;

describe('<RatingStars />', () => {
  it('renders with value of 50', () => {
    const wrapper = shallow(
      <RatingStars value={50} />
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(StarIcon).length).toBe(numEmptyStars + 2);
    expect(wrapper.find(StarHalfIcon).length).toBe(1);
  });

  it('renders with value of 0', () => {
    const wrapper = shallow(
      <RatingStars value={0} />
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(StarIcon).length).toBe(numEmptyStars);
    expect(wrapper.find(StarHalfIcon).length).toBe(0);
  });

  it('renders with value of 100', () => {
    const wrapper = shallow(
      <RatingStars value={100} />
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(StarIcon).length).toBe(numEmptyStars + 5);
    expect(wrapper.find(StarHalfIcon).length).toBe(0);
  });

  it('renders as non-form element', () => {
    const wrapper = shallow(
      <RatingStars
        value={60}
        isFormElement={false}
      />
    );

    expect(wrapper).toMatchSnapshot();
    const firstStarWrapper = wrapper.find(StarIcon).first().parent('div');
    const lastStarWrapper = wrapper.find(StarIcon).last().parent('div');
    expect(firstStarWrapper.prop('role')).not.toBeDefined();
    expect(firstStarWrapper.prop('onClick')).not.toBeDefined();
    expect(lastStarWrapper.prop('role')).not.toBeDefined();
    expect(lastStarWrapper.prop('onClick')).not.toBeDefined();
  });

  it('renders as form element', () => {
    const wrapper = shallow(
      <RatingStars
        value={60}
        isFormElement
      />
    );

    const firstStarWrapper = wrapper.find(StarIcon).first().parent('div');
    const lastStarWrapper = wrapper.find(StarIcon).last().parent('div');

    expect(wrapper).toMatchSnapshot();
    expect(firstStarWrapper.prop('role')).toEqual('button');
    expect(typeof firstStarWrapper.prop('onClick')).toEqual('function');
    expect(lastStarWrapper.prop('role')).toEqual('button');
    expect(typeof lastStarWrapper.prop('onClick')).toEqual('function');
  });

  it('sets a value on filled star click', (done) => {
    const wrapper = mount(
      <RatingStars
        value={100}
        onSelection={(e) => {
          expect(e.target.value).toEqual(20);
          done();
        }}
        isFormElement
      />
    );

    // Get the first filled star
    const firstStarWrapper = wrapper
      .findWhere(comp => comp.name() === 'div' && comp.key() === `${numEmptyStars + 1}`)
      .first();
    expect(wrapper).toMatchSnapshot();
    expect(firstStarWrapper.prop('role')).toEqual('button');
    expect(typeof firstStarWrapper.prop('onClick')).toEqual('function');
    firstStarWrapper.simulate('click', { target: {} });
  });

  it('sets value on empty star click', (done) => {
    const wrapper = shallow(
      <RatingStars
        value={0}
        onSelection={(e) => {
          expect(e.target.value).toEqual(100);
          done();
        }}
        isFormElement
      />
    );

    // Get the empty star at last position
    const lastStarWrapper = wrapper
      .findWhere(comp => comp.name() === 'div' && comp.key() === `${numEmptyStars}`)
      .first();

    expect(wrapper).toMatchSnapshot();
    expect(lastStarWrapper.prop('role')).toEqual('button');
    expect(typeof lastStarWrapper.prop('onClick')).toEqual('function');
    lastStarWrapper.simulate('click', { target: {} });
  });

  it('should only update if value changed', () => {
    const wrapper = shallow(
      <RatingStars value={60} />
    );

    const shouldUpdate = wrapper.instance().shouldComponentUpdate({ value: 20 }, {});
    expect(shouldUpdate).toBe(true);

    const shouldNotUpdate = wrapper.instance().shouldComponentUpdate({ value: 60 }, {});
    expect(shouldNotUpdate).toBe(false);
  });
});
