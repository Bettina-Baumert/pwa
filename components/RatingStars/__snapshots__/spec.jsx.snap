// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RatingStars /> renders as form element 1`] = `
<div
  className="css-13azwyo "
>
  <div
    className="css-4s85e9"
  >
    <div
      className="css-kdb68h css-q4owmg"
      key="1"
      onClick={[Function]}
      role="button"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="2"
      onClick={[Function]}
      role="button"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="3"
      onClick={[Function]}
      role="button"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="4"
      onClick={[Function]}
      role="button"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="5"
      onClick={[Function]}
      role="button"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
  </div>
  <div
    className="css-16y9zin"
  >
    <div
      className="css-kdb68h css-q4owmg"
      key="6"
      onClick={[Function]}
      role="button"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="7"
      onClick={[Function]}
      role="button"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="8"
      onClick={[Function]}
      role="button"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
  </div>
</div>
`;

exports[`<RatingStars /> renders as non-form element 1`] = `
<div
  className="css-13azwyo "
>
  <div
    className="css-4s85e9"
  >
    <div
      className="css-kdb68h css-q4owmg"
      key="1"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="2"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="3"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="4"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="5"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
  </div>
  <div
    className="css-16y9zin"
  >
    <div
      className="css-kdb68h css-q4owmg"
      key="6"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="7"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="8"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
  </div>
</div>
`;

exports[`<RatingStars /> renders with value of 0 1`] = `
<div
  className="css-13azwyo "
>
  <div
    className="css-4s85e9"
  >
    <div
      className="css-kdb68h css-q4owmg"
      key="1"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="2"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="3"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="4"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="5"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
  </div>
  <div
    className="css-16y9zin"
  />
</div>
`;

exports[`<RatingStars /> renders with value of 50 1`] = `
<div
  className="css-13azwyo "
>
  <div
    className="css-4s85e9"
  >
    <div
      className="css-kdb68h css-q4owmg"
      key="1"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="2"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="3"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="4"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="5"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
  </div>
  <div
    className="css-16y9zin"
  >
    <div
      className="css-kdb68h css-q4owmg"
      key="6"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="7"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="2"
    >
      <StarHalfIcon
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
  </div>
</div>
`;

exports[`<RatingStars /> renders with value of 100 1`] = `
<div
  className="css-13azwyo "
>
  <div
    className="css-4s85e9"
  >
    <div
      className="css-kdb68h css-q4owmg"
      key="1"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="2"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="3"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="4"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="5"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
  </div>
  <div
    className="css-16y9zin"
  >
    <div
      className="css-kdb68h css-q4owmg"
      key="6"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="7"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="8"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="9"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="10"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
  </div>
</div>
`;

exports[`<RatingStars /> sets a value on filled star click 1`] = `
<RatingStars
  className=""
  display="small"
  isFormElement={true}
  onSelection={[Function]}
  value={100}
>
  <div
    className="css-13azwyo "
  >
    <div
      className="css-4s85e9"
    >
      <div
        className="css-kdb68h css-q4owmg"
        key="1"
        onClick={[Function]}
        role="button"
      >
        <Star
          className=""
          color={null}
          size="1em"
          viewBox="0 0 24 24"
        >
          <Icon
            className=""
            color={null}
            content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
            size="1em"
            viewBox="0 0 24 24"
          >
            <svg
              className="css-yys9hb "
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                }
              }
              style={
                Object {
                  "fill": null,
                  "fontSize": "1em",
                }
              }
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            />
          </Icon>
        </Star>
      </div>
      <div
        className="css-kdb68h css-q4owmg"
        key="2"
        onClick={[Function]}
        role="button"
      >
        <Star
          className=""
          color={null}
          size="1em"
          viewBox="0 0 24 24"
        >
          <Icon
            className=""
            color={null}
            content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
            size="1em"
            viewBox="0 0 24 24"
          >
            <svg
              className="css-yys9hb "
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                }
              }
              style={
                Object {
                  "fill": null,
                  "fontSize": "1em",
                }
              }
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            />
          </Icon>
        </Star>
      </div>
      <div
        className="css-kdb68h css-q4owmg"
        key="3"
        onClick={[Function]}
        role="button"
      >
        <Star
          className=""
          color={null}
          size="1em"
          viewBox="0 0 24 24"
        >
          <Icon
            className=""
            color={null}
            content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
            size="1em"
            viewBox="0 0 24 24"
          >
            <svg
              className="css-yys9hb "
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                }
              }
              style={
                Object {
                  "fill": null,
                  "fontSize": "1em",
                }
              }
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            />
          </Icon>
        </Star>
      </div>
      <div
        className="css-kdb68h css-q4owmg"
        key="4"
        onClick={[Function]}
        role="button"
      >
        <Star
          className=""
          color={null}
          size="1em"
          viewBox="0 0 24 24"
        >
          <Icon
            className=""
            color={null}
            content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
            size="1em"
            viewBox="0 0 24 24"
          >
            <svg
              className="css-yys9hb "
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                }
              }
              style={
                Object {
                  "fill": null,
                  "fontSize": "1em",
                }
              }
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            />
          </Icon>
        </Star>
      </div>
      <div
        className="css-kdb68h css-q4owmg"
        key="5"
        onClick={[Function]}
        role="button"
      >
        <Star
          className=""
          color={null}
          size="1em"
          viewBox="0 0 24 24"
        >
          <Icon
            className=""
            color={null}
            content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
            size="1em"
            viewBox="0 0 24 24"
          >
            <svg
              className="css-yys9hb "
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                }
              }
              style={
                Object {
                  "fill": null,
                  "fontSize": "1em",
                }
              }
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            />
          </Icon>
        </Star>
      </div>
    </div>
    <div
      className="css-16y9zin"
    >
      <div
        className="css-kdb68h css-q4owmg"
        key="6"
        onClick={[Function]}
        role="button"
      >
        <Star
          className=""
          color={null}
          size="1em"
          viewBox="0 0 24 24"
        >
          <Icon
            className=""
            color={null}
            content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
            size="1em"
            viewBox="0 0 24 24"
          >
            <svg
              className="css-yys9hb "
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                }
              }
              style={
                Object {
                  "fill": null,
                  "fontSize": "1em",
                }
              }
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            />
          </Icon>
        </Star>
      </div>
      <div
        className="css-kdb68h css-q4owmg"
        key="7"
        onClick={[Function]}
        role="button"
      >
        <Star
          className=""
          color={null}
          size="1em"
          viewBox="0 0 24 24"
        >
          <Icon
            className=""
            color={null}
            content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
            size="1em"
            viewBox="0 0 24 24"
          >
            <svg
              className="css-yys9hb "
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                }
              }
              style={
                Object {
                  "fill": null,
                  "fontSize": "1em",
                }
              }
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            />
          </Icon>
        </Star>
      </div>
      <div
        className="css-kdb68h css-q4owmg"
        key="8"
        onClick={[Function]}
        role="button"
      >
        <Star
          className=""
          color={null}
          size="1em"
          viewBox="0 0 24 24"
        >
          <Icon
            className=""
            color={null}
            content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
            size="1em"
            viewBox="0 0 24 24"
          >
            <svg
              className="css-yys9hb "
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                }
              }
              style={
                Object {
                  "fill": null,
                  "fontSize": "1em",
                }
              }
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            />
          </Icon>
        </Star>
      </div>
      <div
        className="css-kdb68h css-q4owmg"
        key="9"
        onClick={[Function]}
        role="button"
      >
        <Star
          className=""
          color={null}
          size="1em"
          viewBox="0 0 24 24"
        >
          <Icon
            className=""
            color={null}
            content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
            size="1em"
            viewBox="0 0 24 24"
          >
            <svg
              className="css-yys9hb "
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                }
              }
              style={
                Object {
                  "fill": null,
                  "fontSize": "1em",
                }
              }
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            />
          </Icon>
        </Star>
      </div>
      <div
        className="css-kdb68h css-q4owmg"
        key="10"
        onClick={[Function]}
        role="button"
      >
        <Star
          className=""
          color={null}
          size="1em"
          viewBox="0 0 24 24"
        >
          <Icon
            className=""
            color={null}
            content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
            size="1em"
            viewBox="0 0 24 24"
          >
            <svg
              className="css-yys9hb "
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                }
              }
              style={
                Object {
                  "fill": null,
                  "fontSize": "1em",
                }
              }
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            />
          </Icon>
        </Star>
      </div>
    </div>
  </div>
</RatingStars>
`;

exports[`<RatingStars /> sets value on empty star click 1`] = `
<div
  className="css-13azwyo "
>
  <div
    className="css-4s85e9"
  >
    <div
      className="css-kdb68h css-q4owmg"
      key="1"
      onClick={[Function]}
      role="button"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="2"
      onClick={[Function]}
      role="button"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="3"
      onClick={[Function]}
      role="button"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="4"
      onClick={[Function]}
      role="button"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
    <div
      className="css-kdb68h css-q4owmg"
      key="5"
      onClick={[Function]}
      role="button"
    >
      <Star
        className=""
        color={null}
        size="1em"
        viewBox="0 0 24 24"
      />
    </div>
  </div>
  <div
    className="css-16y9zin"
  />
</div>
`;
